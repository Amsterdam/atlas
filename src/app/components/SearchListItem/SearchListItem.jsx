import React from 'react'
import RouterLink from 'redux-first-router-link'
import PropTypes from 'prop-types'

const STATUS_OBJECT_GEVORMD = 18

const isString = value => typeof value === 'string'

function isObject(value) {
  return value !== null && typeof value === 'object'
}

const showSubtype = (categorySlug, link) =>
  isString(link.subtype) &&
  (categorySlug === 'ligplaats' ||
    categorySlug === 'standplaats' ||
    (categorySlug === 'openbareruimte' && link.subtype !== 'weg') ||
    (categorySlug === 'adres' && link.subtype !== 'verblijfsobject') ||
    categorySlug === 'gebied' ||
    categorySlug === 'explosief' ||
    (categorySlug === 'monument' && link.subtype === 'complex'))

const getExtraInfo = result => {
  let extraInfo = ''
  if (result.hoofdadres === false) {
    extraInfo += ' (nevenadres)'
  }

  if (isObject(result.vbo_status) && Number(result.vbo_status.code) === STATUS_OBJECT_GEVORMD) {
    extraInfo += ` (${result.vbo_status.omschrijving.toLowerCase()})`
  }

  return extraInfo
}

const SearchListItem = ({ result, category }) => (
  <li>
    <RouterLink className="o-btn o-btn--link qa-list-item-link" to={result.linkTo}>
      {result.label}
    </RouterLink>

    <span className="qa-search-results__link-extra-info">{getExtraInfo(result)}</span>

    {showSubtype(category.slug, result) ? (
      <span className="qa-subtype">
        &nbsp;(
        {result.subtypeLabel})
      </span>
    ) : null}
  </li>
)

SearchListItem.propTypes = {
  result: PropTypes.shape({
    label: PropTypes.string,
    subtype: PropTypes.string,
    subtypeLabel: PropTypes.string,
    linkTo: PropTypes.shape(),
  }).isRequired,
  category: PropTypes.shape().isRequired,
}

export default SearchListItem
