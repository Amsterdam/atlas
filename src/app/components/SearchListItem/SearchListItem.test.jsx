import React from 'react'
import { shallow } from 'enzyme'
import SearchListItem from './SearchListItem'

describe('SearchListItem', () => {
  it('should render straatnamen type weg', () => {
    const props = {
      result: {
        label: 'label',
        subtype: 'weg',
        subtypeLabel: 'weg',
        linkTo: {},
      },
      category: {
        slug: 'straatnamen',
      },
    }

    const component = shallow(<SearchListItem {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('should render openbareruimte type water', () => {
    const props = {
      result: {
        label: 'label',
        subtype: 'water',
        subtypeLabel: 'water',
        linkTo: {},
      },
      category: {
        slug: 'openbareruimte',
      },
    }

    const component = shallow(<SearchListItem {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('should render address type verblijfsobject', () => {
    const props = {
      result: {
        label: 'label',
        subtype: 'verblijfsobject',
        subtypeLabel: 'verblijfsobject',
        linkTo: {},
        hoofdadres: false,
        vbo_status: {
          code: '18',
          omschrijving: 'Verblijfsobject gevormd',
        },
      },
      category: {
        slug: 'adres',
      },
    }

    const component = shallow(<SearchListItem {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('should render monument type complex', () => {
    const props = {
      result: {
        label: 'label',
        subtype: 'complex',
        subtypeLabel: 'complex',
        linkTo: {},
      },
      category: {
        slug: 'monument',
      },
    }

    const component = shallow(<SearchListItem {...props} />)
    expect(component).toMatchSnapshot()
  })
})
