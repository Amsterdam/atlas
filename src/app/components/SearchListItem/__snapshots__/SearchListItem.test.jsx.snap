// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchListItem should render address type verblijfsobject 1`] = `
<li>
  <Connect(Link)
    className="o-btn o-btn--link qa-list-item-link"
    to={Object {}}
  >
    label
  </Connect(Link)>
  <span
    className="qa-search-results__link-extra-info"
  >
     (nevenadres) (verblijfsobject gevormd)
  </span>
</li>
`;

exports[`SearchListItem should render monument type complex 1`] = `
<li>
  <Connect(Link)
    className="o-btn o-btn--link qa-list-item-link"
    to={Object {}}
  >
    label
  </Connect(Link)>
  <span
    className="qa-search-results__link-extra-info"
  />
  <span
    className="qa-subtype"
  >
     (
    complex
    )
  </span>
</li>
`;

exports[`SearchListItem should render openbareruimte type water 1`] = `
<li>
  <Connect(Link)
    className="o-btn o-btn--link qa-list-item-link"
    to={Object {}}
  >
    label
  </Connect(Link)>
  <span
    className="qa-search-results__link-extra-info"
  />
  <span
    className="qa-subtype"
  >
     (
    water
    )
  </span>
</li>
`;

exports[`SearchListItem should render straatnamen type weg 1`] = `
<li>
  <Connect(Link)
    className="o-btn o-btn--link qa-list-item-link"
    to={Object {}}
  >
    label
  </Connect(Link)>
  <span
    className="qa-search-results__link-extra-info"
  />
</li>
`;
